src/pages/index.js

```diff
@@ -0,0 +1,335 @@
+      return;
+    }
+    cardToDelete = cardElement;
+    openModal(deleteCardModal);
+  }
+
+  if (confirmDeleteButton) {
+    confirmDeleteButton.addEventListener("click", () => {
+      if (!cardToDelete) return;
+
+      api
+        .removeCard(cardToDelete.dataset.cardId)
+        .then(() => {
+          cardToDelete.remove();
+          closeModal(deleteCardModal);
+        })
+        .catch(console.error);
+    });
+  }
+
+  if (profileEditButton) {
+    profileEditButton.addEventListener("click", () => {
+      resetForm(profileForm, settings);
+      inputProfileName.value = profileName?.textContent.trim() || "";
+      inputProfileDescription.value =
+        profileDescription?.textContent.trim() || "";
+      openModal(profileEditModal);
+    });
+  }
+
+  if (profileForm) {
+    profileForm.addEventListener("submit", (event) => {
+      event.preventDefault();
+
+      const saveButton = profileForm.querySelector(".modal__submit-btn");
+      const originalText = saveButton.textContent;
+
+      saveButton.textContent = "Saving...";
+      saveButton.disabled = true;
+
+      api
+        .updateUserInfo({
+          name: inputProfileName.value.trim(),
+          about: inputProfileDescription.value.trim(),
+        })
+        .then(() => {
+          if (profileName && inputProfileName) {
+            profileName.textContent = inputProfileName.value.trim();
+          }
+          if (profileDescription && inputProfileDescription) {
+            profileDescription.textContent =
+              inputProfileDescription.value.trim();
+          }
+          closeModal(profileEditModal);
+        })
+        .catch(console.error)
+        .finally(() => {
+          saveButton.textContent = originalText;
+          saveButton.disabled = false;
+        });
+    });
+  }
+
+  if (profileAvatarEditButton) {
+    profileAvatarEditButton.addEventListener("click", () => {
+      resetForm(avatarForm, settings);
+      openModal(editAvatarModal);
+    });
+  }
+
+  if (avatarForm) {
+    avatarForm.addEventListener("submit", (event) => {
+      event.preventDefault();
+      const newAvatarUrl = inputAvatarUrl?.value.trim();
+      if (!newAvatarUrl) return;
+
+      // Show loading state
+      const submitButton = avatarForm.querySelector('button[type="submit"]');
+      if (submitButton) {
+        submitButton.textContent = "Saving...";
+        submitButton.disabled = true;
+      }
+
+      // Make the API call
+      api
+        .updateProfileAvatar(newAvatarUrl)
+        .then((userData) => {
+          if (profileAvatar) {
+            profileAvatar.src = userData.avatar;
+          }
+          closeModal(editAvatarModal);
+          avatarForm.reset();
+        })
+        .catch((error) => {
+          console.error("Error updating avatar:", error);
+        })
+        .finally(() => {
+          if (submitButton) {
+            submitButton.textContent = "Save";
+            submitButton.disabled = false;
```

Comment:
`            submitButton.disabled = false;
`

This makes problems for disabling the button in `.then`. 


If you want to enable the button in case of a server error, then you should do it in the `.catch` block


```js
        .catch((error) => {
          console.error("Error updating avatar:", error);
          submitButton.disabled = false;  // here
        })
        .finally(() => {
          if (submitButton) {
            submitButton.textContent = "Save";
          }
        });

```

**Please, correct it everywhere in the code**